### ISSUE TITLE
Make tab key behaviour more intuitive

### ISSUE THREAD

**Issue by cpcallen** [rachel-fenichel/blockly-keyboard-experimentation#101]
This bug tracks several proposed changes to how the tab key and keyboard navigation interact.

* [x] Make keyboard navigation active any time the workspace (or any part of it) has focus.
  * At the moment it is necessary to tab to the workspace and then press enter to being interacting with it, and then press esc to deactivate keyboard navigation before it is possible to tab away from the workspace.
  * Instead, tabbing to the workspace should suffice to activate navigation, and tabbing away should deactivate it (and be possible at any time without first pressing esc).
* [x] Make it possible to tab to the toolbox.
* [ ] Implement [roving tab index](https://opensource.adobe.com/spectrum-web-components/tools/roving-tab-index/), for better compatibility with screen readers.

Instead, keyboard navigation should be active any time the workspace (or any part of it) has focus.

See also:

* #62
* #63
* #89
* #92
* #96

**Comment by BenHenning** [comment]
So one question: how should navigation to toolbox work? The natural flow per the page layout, I think, would be that the toolbox is tabbed to before the workspace. However, this may not be an ideal flow since the toolbox is only sometimes needed by the user so to me it seems clearer to navigate first to the workspace. Or, are there other thoughts on how the toolbox should be tab-navigable?

**Comment by rachel-fenichel** [comment]
I [summarized our discussion here](https://github.com/google/blockly-keyboard-experimentation/discussions/62#discussioncomment-11268064).

In testing, the kids did not at first know about tabbing, but picked it up relatively quickly and expected it to move logically through the page. I think it's fine for them to have to go to the toolbox and then to the block editor, and that skipping it would make it harder for them to realize that the toolbox is reachable through tabbing.

**Comment by rachel-fenichel** [comment]
Closed because this large-scale feature is now tracked in smaller issues and feature requests.
